src/coreclr/src/binder/assemblybinder.cpp

```diff
@@ -396,9 +396,9 @@ namespace BINDER_SPACE
         BundleFileLocation bundleFileLocation = Bundle::ProbeAppBundle(sCoreLibName, /*pathIsBundleRelative */ true);
         if (!bundleFileLocation.IsValid())
         {
-            sCoreLib.Set(systemDirectory);
             pathSource = BinderTracing::PathSource::ApplicationAssemblies;
         }
+        sCoreLib.Set(systemDirectory);
```

Comment:
what is the `systemDirectory` in the case of a single-file app?

Reply:
It's the path to the `.exe`. That's what the "discussion" is mostly about - internally all these paths are not correct - there are no files there. Typically, unless somebody puts `System.Private.CoreLib.dll` next to the single-file .exe then it would point to a real file - that's why I was looking for usages of this path internally - as far as I can tell we don't actually look at the file at that path anywhere - so it should be fine.

This makes it basically consistent with all the other assemblies which already used the path to the single-file location.